sentry/src/main/java/io/sentry/DateUtils.java

```diff
@@ -114,32 +90,32 @@ private DateUtils() {}
       final long seconds = Long.parseLong(times[0]);
       final long millis = times.length > 1 ? Long.parseLong(times[1]) : 0;
 
-      return getDateTime(new Date((seconds * 1000) + millis));
+      return getDateTime((seconds * 1000) + millis);
     } catch (NumberFormatException e) {
       throw new IllegalArgumentException("timestamp is not millis format " + timestamp);
     }
   }
 
   /**
-   * Get date formatted as expected by Sentry.
+   * Get the UTC/ISO 8601 timestamp from Date
    *
-   * @param date already UTC format
-   * @return the ISO formatted date with millis precision.
+   * @param date the UTC Date
+   * @return the UTC/ISO 8601 timestamp
    */
   public static @NotNull String getTimestamp(final @NotNull Date date) {
-    final DateFormat df = SDF_ISO_FORMAT_WITH_MILLIS.get();
+    final DateFormat df = SDF_ISO_FORMAT_WITH_MILLIS_UTC.get();
     return df.format(date);
   }
 
   /**
-   * Converts the given Date and time to UTC timezone
+   * Get the Date from millis timestamp
    *
-   * @param date the Date with local timezone
-   * @return the Date UTC timezone
+   * @param millis the UTC millis from the epoch
+   * @return the UTC Date
    */
-  public static @NotNull Date getDateTime(final @NotNull Date date)
-      throws IllegalArgumentException {
-    final String timestampIsoFormat = getTimestampIsoFormat(date);
-    return getDateTime(timestampIsoFormat);
+  public static @NotNull Date getDateTime(final long millis) {
```

Comment:
`m`: Why not just use `new Date(millis)`? No need to create an instance of `Calendar` I think. We could remove this method I think or am I missing something?

Reply:
`Date` is not TimeZone aware, we need a UTC Date

Reply:
ps* Calendar actually creates a new Date(), but the UTC ctor is deprecated, and ask to use a Calendar
https://docs.oracle.com/javase/7/docs/api/java/util/Date.html#UTC(int,%20int,%20int,%20int,%20int,%20int)